ncm-opennebula/src/main/perl/opennebula.pm

```diff
@@ -71,7 +119,12 @@ sub process_template
         $self->error("TT processing of $type_rel failed: $tpl->{fail}");
         return;
     }
-    return $tpl;
+
+    if(exists($fw_types{$type_name})) {
```

Comment:
if this is the only usage of `fw_types`, you can do a `if (grep {$_ eq $type_name} @FILEWRITER_TEMPLATES)` here, and not use `fw_types`